cdap-ui/app/hydrator/routes.js

```diff
@@ -128,6 +112,9 @@ angular.module(PKG.name + '.feature.hydrator')
               });
               return defer.promise;
             },
+            rValidNamespace: function($stateParams, myNamespace, myHelpers){
+              myHelpers.validNamespaceResolver($stateParams, myNamespace);
```

Comment:
can you check how many times this function gets called when you go to either studio or detail page? I think all you need to do is add it to the parent `hydrator` state.

Reply:
For some reason `hydrator` state resolvers are not called when I navigate from detail page to studio page. This is the root cause of the bug.